src/IssueViz.Security/Taint/ServerSentEvents/TaintServerEventsListener.cs

```diff
@@ -73,30 +73,37 @@ public async Task ListenAsync()
 
             while (!cancellationTokenSource.IsCancellationRequested)
             {
-                var taintServerEvent = await taintServerEventSource.GetNextEventOrNullAsync();
-
-                switch (taintServerEvent)
+                try
                 {
-                    case null:
-                    {
-                        return;
-                    }
-                    case ITaintVulnerabilityClosedServerEvent taintClosedEvent:
-                    {
-                        taintStore.Remove(taintClosedEvent.Key);
-                        break;
-                    }
-                    case ITaintVulnerabilityRaisedServerEvent taintRaisedEvent:
-                    {
-                        await AddToStoreIfOnTheRightBranchAsync(taintRaisedEvent);
-                        break;
-                    }
-                    default:
+                    var taintServerEvent = await taintServerEventSource.GetNextEventOrNullAsync();
+
+                    switch (taintServerEvent)
                     {
-                        logger.LogVerbose($"[TaintServerEventsListener] Unrecognized taint event type: {taintServerEvent.GetType()}");
-                        break;
+                        case null:
+                        {
+                            return;
```

Comment:
Unrelated: might be worth commenting when this is expected (I've forgotten already - ?stream is disposed?)